target_chains/ethereum/contracts/forge-test/Pyth.WormholeMerkleAccumulator.t.sol

```diff
@@ -421,7 +590,82 @@ contract PythWormholeMerkleAccumulatorTest is
             priceFeedMessages
         );
 
+        bytes32[] memory priceIds = new bytes32[](numPriceFeeds);
+        for (uint i = 0; i < numPriceFeeds; i++) {
+            priceIds[i] = priceFeedMessages[i].priceId;
+        }
         vm.expectRevert(PythErrors.InsufficientFee.selector);
         pyth.updatePriceFeeds{value: 0}(updateData);
+
+        vm.expectRevert(PythErrors.InsufficientFee.selector);
+        pyth.parsePriceFeedUpdates{value: 0}(
+            updateData,
+            priceIds,
+            0,
+            MAX_UINT64
+        );
+    }
+
+    function testParsePriceFeedWithWormholeMerkleWorks(uint seed) public {
+        setRandSeed(seed);
+
+        uint numPriceFeeds = (getRandUint() % 10) + 1;
+        PriceFeedMessage[]
+            memory priceFeedMessages = generateRandomPriceFeedMessage(
+                numPriceFeeds
+            );
+        (
+            bytes[] memory updateData,
+            uint updateFee
+        ) = createWormholeMerkleUpdateData(priceFeedMessages);
+        bytes32[] memory priceIds2 = new bytes32[](numPriceFeeds);
+        for (uint i = 0; i < numPriceFeeds; i++) {
+            priceIds2[i] = priceFeedMessages[i].priceId;
+        }
+        PythStructs.PriceFeed[] memory priceFeeds2 = pyth.parsePriceFeedUpdates{
+            value: updateFee
+        }(updateData, priceIds2, 0, MAX_UINT64);
+
+        for (uint i = 0; i < priceFeeds2.length; i++) {
+            assertParsedPriceFeed(
+                priceFeeds2[i],
+                priceFeedMessages[i],
+                priceIds2[i]
+            );
+        }
+
+        // update priceFeedMessages
+        for (uint i = 0; i < numPriceFeeds; i++) {
+            priceFeedMessages[i].price = getRandInt64();
+            priceFeedMessages[i].conf = getRandUint64();
+            priceFeedMessages[i].expo = getRandInt32();
+
+            // Increase the publish time if it is not causing an overflow
+            if (priceFeedMessages[i].publishTime != type(uint64).max) {
+                priceFeedMessages[i].publishTime += 1;
```

Comment:
I think you should set the publishTime to be random -- it literally doesn't matter whether or not this is more recent.